### PR TITLE
feat(ops):   check dns resolver address is valid

### PR DESCRIPTION
**Pull request by tzssangglass**
### Description

Fixes # (issue)

### Checklist

- [x] I have explained the need for this PR and the problem it solves
- [ ] I have explained the changes or the new features added to this PR
- [x] I have added tests corresponding to this change
- [ ] I have updated the documentation to reflect this change
- [ ] I have verified that this change is backward compatible (If not, please discuss on the [APISIX mailing list](https://github.com/apache/apisix/tree/master#community) first)

### CODE REVIEW

File: apisix/cli/ops.lua [hunk 1]
```diff
@@ -602,6 +602,10 @@ Please modify "admin_key" in conf/config.yaml .
     end
 
     for i, r in ipairs(sys_conf["dns_resolver"]) do
+        local res, err = cli_ip.valid(r)
+        if not res then
+            util.die("invalid dns resolver address: ", r,  ", err: ", err)
```
Comment (spacewander) [inline apisix/cli/ops.lua:607]: Would be better to ignore the unexpected resolver instead of dying of it?
And we can add a warn log for it.

Comment (tzssangglass) [reply apisix/cli/ops.lua]: agree, fix it.

File: apisix/cli/ops.lua [hunk 2]
```diff
@@ -611,6 +612,14 @@ Please modify "admin_key" in conf/config.yaml .
                 sys_conf["dns_resolver"][i] = "[" .. r .. "]"
             end
         end
+
+        -- check if the dns_resolver is ipv6 address with zone_id
+        -- Nnginx does not support this form
+        if r:find("%%") then
+            print("invalid dns resolver address: " .. r ..
```
Comment (spacewander) [inline apisix/cli/ops.lua:619]: Please use `stderr:write` and watch out for the indent.

Comment (spacewander) [reply apisix/cli/ops.lua:619]: Call it unsupported DNS resolver would be better?

File: t/cli/test_dns.sh [hunk 3]
```diff
@@ -96,3 +96,25 @@ if ! grep "resolver 127.0.0.1 valid=30 ipv6=off;" conf/nginx.conf > /dev/null; t
...
+' > conf/config.yaml
+
+out=$(make init 2>&1 || true)
+
+if ! echo "$out" | grep "invalid dns resolver address"; then
+    echo "failed: should check dns resolver address is valid"
+    exit 1
+fi
+
+if ! grep "resolver 127.0.0.1 ipv6=on;" conf/nginx.conf > /dev/null; then
```
Comment (spacewander) [inline t/cli/test_dns.sh:115]: Should check `fe80::21c:42ff:fe00:18%eth0`'s absence too.

**Review by spacewander — DISMISSED** [review]

File: apisix/cli/ops.lua [hunk 4]
```diff
@@ -611,6 +612,14 @@ Please modify "admin_key" in conf/config.yaml .
                 sys_conf["dns_resolver"][i] = "[" .. r .. "]"
             end
         end
+
+        -- check if the dns_resolver is ipv6 address with zone_id
+        -- Nnginx does not support this form
```
Comment (bzp2010) [inline apisix/cli/ops.lua:617]:
```suggestion
        -- Nginx does not support this form
```

**Review by soulbird — APPROVED** [review]

File: apisix/cli/ops.lua (hunk 4 above)
Comment (tzssangglass) [reply apisix/cli/ops.lua]: fixed

**Review by bzp2010 — APPROVED** [review]

**Review by spacewander — APPROVED** [review]

**Pull request merged**